library/core/src/main/java/com/google/android/exoplayer2/extractor/ts/Ac3Reader.java

```diff
@@ -188,18 +192,36 @@ private boolean skipToNextSync(ParsableByteArray pesBuffer) {
   private void parseHeader() {
     headerScratchBits.setPosition(0);
     Ac3Util.Ac3SyncFrameInfo frameInfo = Ac3Util.parseAc3SyncframeInfo(headerScratchBits);
-    if (format == null || frameInfo.channelCount != format.channelCount
-        || frameInfo.sampleRate != format.sampleRate
-        || frameInfo.mimeType != format.sampleMimeType) {
+    if ((format == null || frameInfo.sampleRate != format.sampleRate
+        || frameInfo.mimeType != format.sampleMimeType)
```

Comment:
Hi, ojw28,

After discuss with my colleagues, all of us think it is difficult to output "I,D" in a single sample based on the current ExoPlayer logic. The reasons is both I frame and D frame share the same sync word "0x0B77". There is no unique sync word to identify the beginning of one access unit. (Unless you parse the header in bitstream.) This is the limitation of EAC3 format. If we want to outputting "I,D" in a single sample, it seems that we need to re-design the state machine in "consume" method. (Maybe you have better idea? :)) This modification is big. 

BTW, one information for your reference. Based on the EAC3 spec, in theory, one access unit can include maximum 8 I frames with 64 D frames, which are I0,D0,D1,...,D7,I1,D0,D1,...,D7,I2,D0,D1,...,D7,...I7,D0,D1,...,D7  
But in reality, I didn't find this kind of stream. In most cases, I think supporting "I,D" is enough. (This is for 7.1 EAC3 stream)

Finally, it is up to you to decide whether accepting our current modification. If NOT, please feel free to let me know and I'll submit a new pull request for other two files modification.

And thanks for your carefully code review and detailed comments. It's truly helpful. 

Yanning